test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/V3DetailControlModelTests.cs

```diff
@@ -198,6 +200,87 @@ public async Task SetCurrentPackageAsync_SortsVersions_ByNuGetVersionDesc()
             Assert.Equal(expectedVersions, actualVersions);
         }
 
+        [Theory]
+        [InlineData(NuGetProjectKind.PackageReference, "*", "2.10.0")]
+        [InlineData(NuGetProjectKind.PackageReference, "1.*", "2.10.0")]
+        [InlineData(NuGetProjectKind.PackageReference, "(0.1,3.4)", "2.10.0")]
+        [InlineData(NuGetProjectKind.PackageReference, "2.10.0", "2.10.0")]
```

Comment:
Can you test whatever happens for non-PackageReference kinds?

If it's an exception, test that! I'd create a separate test. 
Search our test code for `Assert.Throws` to see examples.